docker/package/wizard_structure.py

```diff
@@ -45,12 +50,31 @@ def _validator(input):
     return _validator
 
 
+def dirpath_validator(input):
+    if input and not os.path.isdir(input):
+        raise ValueError("Please input a valid path to a directory.")
+    return input
+
+
 def filepath_validator(input):
     if input and not os.path.isfile(input):
         raise ValueError("Please input a valid file path.")
     return input
 
 
+def reachable_url_validator(suffix=None):
+    def _validator(input):
+        if input:
+            try:
+                print(urllib.parse.urljoin(input, suffix))
+                urllib.request.urlopen(urllib.parse.urljoin(input, suffix))
+            except (urllib.error.URLError, ValueError):
+                raise ValueError("Please input a reachable URL.")
```

Comment:
I think error should inform why we treat provided endoint as invalid
```suggestion
                raise ValueError(f"{urllib.parse.urljoin(input, suffix)} is unreachable. Please input valid node endpoint URL.")
```

Reply:
It's a validator for any reachable URLs though, e.g. it's also used in the setup wizard now to validate snapshot URLs. I think it's worth it to preserve generality in the validator and just depend on the step prompt. Printing the full suffix URL sounds good though